src/OSPSuite.Core/Domain/Services/MoleculePropertiesContainerTask.cs

```diff
@@ -53,15 +56,24 @@ public IContainer NeighborhoodMoleculeContainerFor(Neighborhood neighborhood, st
          return moleculeContainer;
       }
 
-      public IContainer CreateGlobalMoleculeContainerFor(IContainer rootContainer, MoleculeBuilder moleculeBuilder, SimulationBuilder simulationBuilder)
+      public IContainer CreateGlobalMoleculeContainerFor(IContainer rootContainer, MoleculeBuilder moleculeBuilder, SimulationBuilder simulationBuilder, ValidationResult validationResult)
       {
          var globalMoleculeContainer = addContainerUnder(rootContainer, moleculeBuilder, moleculeBuilder.Name, simulationBuilder)
             .WithContainerType(ContainerType.Molecule);
 
          //Add global molecule dependent parameters
          if (moleculeBuilder.IsFloatingXenobiotic)
          {
-            simulationBuilder.SpatialStructures.Select(x => x.GlobalMoleculeDependentProperties).Each(x => { globalMoleculeContainer.AddChildren(addAllParametersFrom(x, simulationBuilder)); });
+            simulationBuilder.SpatialStructures
+               .Select(x => x.GlobalMoleculeDependentProperties)
```

Comment:
instead of this select and then SelctMany, I could just select the last `GlobalMoleculeDependentProperties` which would be similar to the what we do with other containers. But I am not sure if this would create a problem ??